File: Leetcode_solutions/Pascal'sTriangleII.js
/**
 * @param {number} rowIndex
 * @return {number[]}
 */

var getRow = function(rowIndex, row = 0, column = 1, output = [[1]]) {
  // i row
  // j column
  // base
  if (rowIndex === 0) {
    return [1];
  }
  if (row > rowIndex) {
    return output[rowIndex];
  } else {
    if (row >= output.length) {
      output.push([]);
      output[row].push(1);
      if (row > 1) {
        for (let i = 1; i < row; i++) {
          output[row].push(output[row - 1][i - 1] + output[row - 1][i]);
        }
      }
      output[row].push(1);
    }
    return getRow(rowIndex, row + 1, column++, output);
  }
};

console.log(getRow(3));

File: Leetcode_solutions/Pascal's Triangle II.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        ans = [[1]]
        for i in range(2, rowIndex + 2):
            current = [1 for _ in range(i)]
            for j in range(len(current)):
                if j == 0 or j == len(current) - 1:
                    current[j] = 1
                else:
                    current[j] = ans[-1][j] + ans[-1][j - 1]
            ans.append(current)
        return ans[rowIndex]
